src/browser/graphics/components/omnibar/Ticker.tsx

```diff
@@ -8,12 +8,14 @@ export function Ticker({ style }: { style: CSSProperties }) {
   const currentComponentIndex = useRef(0)
   const currentComponentContainer = useRef<HTMLSpanElement | null>(null)
   const [timestamp, setTimestamp] = useState(Date.now())
+  const tickerContainerRef = useRef<HTMLDivElement | null>(null)
+  const [needsScrolling, setNeedsScrolling] = useState(false)
 
   const messageTypes: React.JSX.Element[] = [
     genericMessage('Welcome to <span class="highlight">GTAMarathon 2025</span>! Enjoy the show!'),
     genericMessage('Check out the merch store over at <span class="highlight">merch.gtamarathon.com</span>!'),
     genericMessage(`Type <span class="highlight">!schedule</span> in the chat to see what's on next!`),
-    <NextRun key={timestamp} time={20} onEnd={showNextElement} />,
+    <NextRun key={timestamp} time={20} onEnd={showNextElement} containerRef={tickerContainerRef} onScrollingNeeded={setNeedsScrolling} />,
```

Comment:
```suggestion
    <NextRun key={currentComponentIndex.current} time={20} onEnd={showNextElement} containerRef={tickerContainerRef} onScrollingNeeded={setNeedsScrolling} />,
```

Should the key here also be updated like in other places?